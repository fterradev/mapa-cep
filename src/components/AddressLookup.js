import React, { PureComponent } from 'react';
import { Segment, Header, Form, Button, Input } from 'semantic-ui-react';
import { lookupAddress } from '../helper';

class AddressLookup extends PureComponent {
  inputCep = React.createRef();
  
  state = {
    address: null,
    cep: '',
    resultPanelIsOpen: false,
    errorMsg: null
  };

  findLocation = () => {
    lookupAddress(this.state.cep)
      .then(address =>
        this.setState({
          address,
          resultPanelIsOpen: true,
          errorMsg: null
        })
      )
      .catch(reason =>
        this.setState({
          errorMsg: reason,
          address: null,
          resultPanelIsOpen: true
        })
      );
  };

  componentDidMount() {
    //this.findLocation();
    this.inputCep.current.focus();
  }

  render() {
    const { cep, address, errorMsg, resultPanelIsOpen } = this.state;
    const { renderResult } = this.props;
    return (
      <div>
        <Segment>
          <Header as="h3">Consultar</Header>
          <Form onSubmit={this.findLocation}>
            <Form.Group>
              <Form.Field inline required>
                <label>CEP</label>
                <Input
                  type="text"
                  ref={this.inputCep}
                  onChange={e => {
                    const input = e.target;
                    if (input.validity.patternMismatch) {
                      input.setCustomValidity('Favor inserir um CEP com 9 dÃ­gitos');
                    } else {
                      input.setCustomValidity('');
                    }
                    this.setState({
                      cep: input.value
                    })
                  }}
                  value={cep}
                  required
                  pattern="[0-9]{5}-?[0-9]{3}"
                />
              </Form.Field>
              <Button type="submit">Buscar</Button>
            </Form.Group>
          </Form>
        </Segment>
        {(resultPanelIsOpen && (
          <Segment>
            <Button
              icon="close"
              floated="right"
              onClick={() =>
                this.setState({
                  resultPanelIsOpen: false
                })
              }
            />
            {address && renderResult(address)}
            {errorMsg ? errorMsg : ''}
          </Segment>
        )) ||
          null}
      </div>
    );
  }
}

export default AddressLookup;
