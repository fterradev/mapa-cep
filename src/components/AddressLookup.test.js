import React from 'react';
import { shallow } from 'enzyme';
import AddressLookup from './AddressLookup';

describe('Wallet', () => {
  const mockRenderResult = jest.fn();
  const addressLookup = shallow(<AddressLookup renderResult={mockRenderResult} />);

  it('renders properly', () => {
    expect(addressLookup).toMatchSnapshot();
  });
});
