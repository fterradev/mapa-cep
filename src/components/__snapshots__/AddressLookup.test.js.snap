// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddressLookup renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AddressLookup
    renderResult={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Segment>
          <Header
                    as="h3"
          >
                    Consultar
          </Header>
          <Form
                    as="form"
                    onSubmit={[Function]}
          >
                    <FormGroup>
                              <FormField
                                        inline={true}
                                        required={true}
                              >
                                        <label>
                                                  CEP
                                        </label>
                                        <Input
                                                  onChange={[Function]}
                                                  pattern="[0-9]{5}-?[0-9]{3}"
                                                  required={true}
                                                  type="text"
                                                  value=""
                                        />
                              </FormField>
                              <Button
                                        as="button"
                                        role="button"
                                        type="submit"
                              >
                                        Buscar
                              </Button>
                    </FormGroup>
          </Form>
</Segment>,
        null,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Header
              as="h3"
>
              Consultar
</Header>,
            <Form
              as="form"
              onSubmit={[Function]}
>
              <FormGroup>
                            <FormField
                                          inline={true}
                                          required={true}
                            >
                                          <label>
                                                        CEP
                                          </label>
                                          <Input
                                                        onChange={[Function]}
                                                        pattern="[0-9]{5}-?[0-9]{3}"
                                                        required={true}
                                                        type="text"
                                                        value=""
                                          />
                            </FormField>
                            <Button
                                          as="button"
                                          role="button"
                                          type="submit"
                            >
                                          Buscar
                            </Button>
              </FormGroup>
</Form>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "as": "h3",
              "children": "Consultar",
            },
            "ref": null,
            "rendered": "Consultar",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "as": "form",
              "children": <FormGroup>
                <FormField
                                inline={true}
                                required={true}
                >
                                <label>
                                                CEP
                                </label>
                                <Input
                                                onChange={[Function]}
                                                pattern="[0-9]{5}-?[0-9]{3}"
                                                required={true}
                                                type="text"
                                                value=""
                                />
                </FormField>
                <Button
                                as="button"
                                role="button"
                                type="submit"
                >
                                Buscar
                </Button>
</FormGroup>,
              "onSubmit": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <FormField
                    inline={true}
                    required={true}
>
                    <label>
                                        CEP
                    </label>
                    <Input
                                        onChange={[Function]}
                                        pattern="[0-9]{5}-?[0-9]{3}"
                                        required={true}
                                        type="text"
                                        value=""
                    />
</FormField>,
                  <Button
                    as="button"
                    role="button"
                    type="submit"
>
                    Buscar
</Button>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": Array [
                      <label>
                        CEP
</label>,
                      <Input
                        onChange={[Function]}
                        pattern="[0-9]{5}-?[0-9]{3}"
                        required={true}
                        type="text"
                        value=""
/>,
                    ],
                    "inline": true,
                    "required": true,
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "CEP",
                      },
                      "ref": null,
                      "rendered": "CEP",
                      "type": "label",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "onChange": [Function],
                        "pattern": "[0-9]{5}-?[0-9]{3}",
                        "required": true,
                        "type": "text",
                        "value": "",
                      },
                      "ref": Object {
                        "current": null,
                      },
                      "rendered": null,
                      "type": [Function],
                    },
                  ],
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "as": "button",
                    "children": "Buscar",
                    "role": "button",
                    "type": "submit",
                  },
                  "ref": null,
                  "rendered": "Buscar",
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      null,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Segment>
            <Header
                        as="h3"
            >
                        Consultar
            </Header>
            <Form
                        as="form"
                        onSubmit={[Function]}
            >
                        <FormGroup>
                                    <FormField
                                                inline={true}
                                                required={true}
                                    >
                                                <label>
                                                            CEP
                                                </label>
                                                <Input
                                                            onChange={[Function]}
                                                            pattern="[0-9]{5}-?[0-9]{3}"
                                                            required={true}
                                                            type="text"
                                                            value=""
                                                />
                                    </FormField>
                                    <Button
                                                as="button"
                                                role="button"
                                                type="submit"
                                    >
                                                Buscar
                                    </Button>
                        </FormGroup>
            </Form>
</Segment>,
          null,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <Header
                as="h3"
>
                Consultar
</Header>,
              <Form
                as="form"
                onSubmit={[Function]}
>
                <FormGroup>
                                <FormField
                                                inline={true}
                                                required={true}
                                >
                                                <label>
                                                                CEP
                                                </label>
                                                <Input
                                                                onChange={[Function]}
                                                                pattern="[0-9]{5}-?[0-9]{3}"
                                                                required={true}
                                                                type="text"
                                                                value=""
                                                />
                                </FormField>
                                <Button
                                                as="button"
                                                role="button"
                                                type="submit"
                                >
                                                Buscar
                                </Button>
                </FormGroup>
</Form>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "as": "h3",
                "children": "Consultar",
              },
              "ref": null,
              "rendered": "Consultar",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "as": "form",
                "children": <FormGroup>
                  <FormField
                                    inline={true}
                                    required={true}
                  >
                                    <label>
                                                      CEP
                                    </label>
                                    <Input
                                                      onChange={[Function]}
                                                      pattern="[0-9]{5}-?[0-9]{3}"
                                                      required={true}
                                                      type="text"
                                                      value=""
                                    />
                  </FormField>
                  <Button
                                    as="button"
                                    role="button"
                                    type="submit"
                  >
                                    Buscar
                  </Button>
</FormGroup>,
                "onSubmit": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": Array [
                    <FormField
                      inline={true}
                      required={true}
>
                      <label>
                                            CEP
                      </label>
                      <Input
                                            onChange={[Function]}
                                            pattern="[0-9]{5}-?[0-9]{3}"
                                            required={true}
                                            type="text"
                                            value=""
                      />
</FormField>,
                    <Button
                      as="button"
                      role="button"
                      type="submit"
>
                      Buscar
</Button>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "children": Array [
                        <label>
                          CEP
</label>,
                        <Input
                          onChange={[Function]}
                          pattern="[0-9]{5}-?[0-9]{3}"
                          required={true}
                          type="text"
                          value=""
/>,
                      ],
                      "inline": true,
                      "required": true,
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "CEP",
                        },
                        "ref": null,
                        "rendered": "CEP",
                        "type": "label",
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {
                          "onChange": [Function],
                          "pattern": "[0-9]{5}-?[0-9]{3}",
                          "required": true,
                          "type": "text",
                          "value": "",
                        },
                        "ref": Object {
                          "current": null,
                        },
                        "rendered": null,
                        "type": [Function],
                      },
                    ],
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "as": "button",
                      "children": "Buscar",
                      "role": "button",
                      "type": "submit",
                    },
                    "ref": null,
                    "rendered": "Buscar",
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        null,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
